golang/cosmos/Makefile

```diff
@@ -73,17 +73,18 @@ compile-gyp:
 	node-gyp configure build $(GYP_DEBUG) || { status=$$?; rm -f binding.gyp; exit $$status; }
 	rm -f binding.gyp
 
-compile-libdaemon: go-mod-cache
+compile-libdaemon: go.sum
 	go build -v $(MOD_READONLY) $(SHARED_BUILD_FLAGS) -buildmode=c-shared \
 		-o build/libagcosmosdaemon.so ./cmd/libdaemon/main.go
 
-go-mod-cache: go.sum
-	@echo "--> Download go modules to local cache"
-	@go mod download
```

Comment:
Keep `go-mod-cache` in the Makefile, do avoid differences vs gaia.

Reply:
Hmm, can you elaborate? AFAICT, cosmos-sdk hasn't had a "go-mod-cache" target since it was removed by https://github.com/cosmos/cosmos-sdk/commit/6c1c2cce0461cfbbf6eb05060e28e7e000d4c906 .

Reply:
Ah, I see: https://github.com/cosmos/gaia/blob/48275fc4c4952e3a365de2ee009db0bed2541699/Makefile#L122

Reply:
Opened a PR there: https://github.com/cosmos/gaia/pull/2915